onnxruntime-sys/build.rs

```diff
@@ -43,8 +55,24 @@ fn main() {
 fn main() {
     let libort_install_dir = prepare_libort_dir();
 
-    let include_dir = libort_install_dir.join("include");
-    let lib_dir = libort_install_dir.join("lib");
+    #[cfg(not(feature = "directml"))]
+    let (include_dir, lib_dir) = (
+        libort_install_dir.join("include"),
+        libort_install_dir.join("lib"),
+    );
+
+    #[cfg(feature = "directml")]
+    let (include_dir, lib_dir) = (
+        libort_install_dir.join("build/native/include"),
+        libort_install_dir
+            .join("runtimes")
+            .join(format!(
+                "{}-{}",
+                TRIPLET.os.as_onnx_str(),
+                TRIPLET.arch.as_onnx_str()
```

Comment:
実際に voicevox_core で使ってみてリンクエラーになり気づいたのですが、`TRIPLET.arch.as_onnx_str()` だと DirectML 版 onnxruntime のディレクトリの命名規則と合わないようです（例えば `win-x64` という名前になってほしいところが `win-x86-64` という名前になってしまう）。以下のように記述したところビルドが通りました。

```suggestion
                match TRIPLET.arch {
                    Architecture::X86 => Cow::from("x86"),
                    Architecture::X86_64 => Cow::from("x64"),
                    Architecture::Arm => Cow::from("arm"),
                    Architecture::Arm64 => Cow::from("arm64"),
                }
```

Reply:
ここx64に書き換えてしまうと今度はほかのダウンロードがうまく行かなくなってしまうのでちょっとかんがえるひつようがありそうですね

Reply:
> ここx64に書き換えてしまうと今度はほかのダウンロードがうまく行かなくなってしまうのでちょっとかんがえるひつようがありそうですね

DirectMLの場合だけ書き換えをすることを例示したつもりだったのですが、他のダウンロードに対してどのような影響がありそうでしょうか？（すみません、私が理解できていないだけかもしれません）

Reply:
@PickledChair 新しく専用の関数を定義する形で修正しました

Reply:
@qwerty2501 なるほど、dll以外のダウンロード内容物に関して操作が必要になったときには関数化されていた方が便利ですね。より良い形だと思いました！